src/transformers/trainer_utils.py

```diff
@@ -46,7 +46,38 @@
     import tensorflow as tf
 
 
-def set_seed(seed: int):
+def seed_worker():
+    """
+    Helper function to set worker seed during Dataloader initialization.
+    """
+    worker_seed = torch.initial_seed() % 2**32
+    set_seed(worker_seed)
+
+
+def enable_determinism_for_distributed_training():
```

Comment:
The idea would be for this function to take `seed` here.
```suggestion
def enable_full_determinism(seed: int):
```
and then call `set_seed` inside (instead of `set_seed` calling this function).

(Also changing the name to be a bit shorter.)